unity-sdk/AndroidPlugin/OptimoveUnityPlugin/src/main/java/com/optimove/unity/plugin/UnityProxy.java

```diff
@@ -1,4 +1,376 @@
+import com.optimove.android.optimobile.InAppInboxItem;
+import com.optimove.android.optimobile.InAppInboxSummary;
+import com.optimove.android.optimobile.OptimoveInApp;
+import com.unity3d.player.UnityPlayer;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.net.URL;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.TimeZone;
+
 public class UnityProxy {
+    private static JSONObject pendingPush;
+
+    public static void setUserId(String userId) {
+        try {
+            Optimove.getInstance().setUserId(userId);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void setUserEmail(String userEmail) {
+        try {
+            Optimove.getInstance().setUserEmail(userEmail);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void reportEvent(String eventName) {
+        try {
+            Optimove.getInstance().reportEvent(eventName);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void reportEvent(String eventName, String eventParamsString) {
+
+        if (eventParamsString == null) {
+            reportEvent(eventName);
+            return;
+        }
+
+        try {
+            JSONObject eventParamsJson = new JSONObject(eventParamsString);
+            Iterator<String> iterator = eventParamsJson.keys();
+            Map<String, Object> eventParams = new HashMap<>();
+            while (iterator.hasNext()) {
+                String key = iterator.next();
+                eventParams.put(key, eventParamsJson.get(key));
+            }
+            Optimove.getInstance().reportEvent(eventName, eventParams);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static void registerUser(String userId, String userEmail) {
+        try {
+            Optimove.getInstance().registerUser(userId, userEmail);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static void reportScreenVisit(String screenName) {
+        try {
+            Optimove.getInstance().reportScreenVisit(screenName);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    public static void reportScreenVisit(String screenName, String screenCategory) {
+        try {
+            Optimove.getInstance().reportScreenVisit(screenName, screenCategory);
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+    }
+
+    public static String getVisitorId() {
+        try {
+
+            return Optimove.getInstance().getVisitorId();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return "";
```

Comment:
better to return null, "" is a value, null is no value

Reply:
done